datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1172,16 +1174,22 @@ fn binary_to_string_coercion(
     match (lhs_type, rhs_type) {
         (Binary, Utf8) => Some(Utf8),
         (Binary, LargeUtf8) => Some(LargeUtf8),
+        (Binary, Utf8View) => Some(Utf8View),
```

Comment:
These rules would allow running `binary like utf8view` -- do we have a usecase for that? We could write tests for it, but I don't think it would be widely used 🤔